packages/flutter/lib/src/material/input.dart

```diff
@@ -16,6 +16,125 @@ import 'theme.dart';
+  final String hintText;
+
+  /// The style to use for the text being edited.
+  final TextStyle style;
+
+  /// Whether to hide the text being edited (e.g., for passwords).
+  ///
+  /// When this is set to true, all the characters in the input are replaced by
+  /// U+2022 BULLET characters (•).
+  final bool hideText;
+
+  /// The maximum number of lines for the text to span, wrapping if necessary.
+  /// If this is 1 (the default), the text will not wrap, but will scroll
+  /// horizontally instead.
+  final int maxLines;
+
+  /// Called when the text being edited changes.
+  ///
+  /// The [value] must be updated each time [onChanged] is invoked.
+  final ValueChanged<InputValue> onChanged;
+
+  /// Called when the user indicates that they are done editing the text in the field.
+  final ValueChanged<InputValue> onSubmitted;
+
+  @override
+  _InputFieldState createState() => new _InputFieldState();
+}
+
+class _InputFieldState extends State<InputField> {
+  GlobalKey<RawInputState> _rawInputKey = new GlobalKey<RawInputState>();
+  GlobalKey<RawInputState> _focusKey = new GlobalKey(debugLabel: "_InputFieldState _focusKey");
+
+  GlobalKey get focusKey => config.focusKey ?? (config.key is GlobalKey ? config.key : _focusKey);
+
+  void requestKeyboard() {
+    _rawInputKey.currentState?.requestKeyboard();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMaterial(context));
+    final InputValue value = config.value ?? InputValue.empty;
+    final ThemeData themeData = Theme.of(context);
+    final TextStyle textStyle = config.style ?? themeData.textTheme.subhead;
+
+    final List<Widget> stackChildren = <Widget>[
+      new GestureDetector(
+        key: focusKey == _focusKey ? _focusKey : null,
+        behavior: HitTestBehavior.opaque,
+        onTap: () {
+          requestKeyboard();
+        },
+        // Since the focusKey may have been created here, defer building the
+        // RawInput until the focusKey's context has been set. This is necessary
+        // because the RawInput will check the focus, like Focus.at(focusContext),
+        // when it builds.
+        child: new Builder(
+          builder: (BuildContext context) {
+            return new RawInput(
+              key: _rawInputKey,
+              value: value,
+              focusKey: focusKey,
+              style: textStyle,
+              hideText: config.hideText,
+              maxLines: config.maxLines,
+              cursorColor: themeData.textSelectionColor,
+              selectionColor: themeData.textSelectionColor,
+              selectionControls: materialTextSelectionControls,
+              platform: Theme.of(context).platform,
+              keyboardType: config.keyboardType,
+              onChanged: config.onChanged,
+              onSubmitted: config.onSubmitted,
+            );
+          }
+        ),
+      ),
+    ];
+
+    if (config.hintText != null && value.text.isEmpty) {
```

Comment:
in particular, I would have expected the hint feature to be in the same class as the underline and label and so forth, rather than being specific to a text field subclass.